earth2grid/latlon.py

```diff
@@ -96,10 +96,19 @@ def __init__(self, src: LatLonGrid, lat: np.ndarray, lon: np.ndarray, cylinder:
         # TODO add device switching logic (maybe use torch registers for this
         # info)
         long = src.lon.ravel()
+        long_neg = (long < 0).any()  # are there any negative longitudes?
+
         if self.cylinder:
-            long = np.concatenate([long, [360]], axis=-1)
+            long = np.concatenate([long, [180 if long_neg else 360]], axis=-1)
         long_t = torch.from_numpy(long)
 
+        # make sure lon_query is in the same range as long
+        lon_query = torch.from_numpy(lon.ravel())
+        if long_neg:
+            lon_query = lon_query % 360 - 180
```

Comment:
won't this shift the grid by 180?

I think you want
```suggestion
            lon_query = (lon_query + 180) % 360 - 180
```

(I may be misunderstanding something though)